tests/core/nogo/custom/return_bool_analyzer.go

```diff
@@ -0,0 +1,37 @@
+// returnboolanalyzer checks for functions that return bool.
+package returnboolanalyzer
+
+import (
+	"go/ast"
+
+	"golang.org/x/tools/go/analysis"
+)
+
+const doc = `report functions that return booleans
+
+The return_bool_analyzer analyzer reports functions that return booleans.`
+
+var Analyzer = &analysis.Analyzer{
+	Name: "return_bool_analyzer",
+	Run:  run,
+	Doc:  doc,
+}
+
+func run(pass *analysis.Pass) (interface{}, error) {
+	for _, f := range pass.Files {
+		ast.Inspect(f, func(n ast.Node) bool {
+			switch n := n.(type) {
+			case *ast.FuncDecl:
+				if results := n.Type.Results; results != nil {
```

Comment:
Not a big deal, but if you're matching types, use the types, not the syntax:

```
results := pass.InfoTypes.Defs[n.Name].Type().(*types.Signature).Results()
for i := 0; i < results.Len(); i++ {
    if results.At(i).Type() == types.Typ[types.Bool] { ... }
}
```

Do you want to match exactly bool, or named boolean types too?

Reply:
Yes, it handles both.